File: src/main/kotlin/io/energyconsumptionoptimizer/mapservice/domain/ports/HouseMapService.kt
package io.energyconsumptionoptimizer.mapservice.domain.ports

import io.energyconsumptionoptimizer.mapservice.domain.HouseMap

/**
 * Service interface for retrieving the complete house map.
 */
interface HouseMapService {
    /**
     * Retrieves the complete house map including floor plan, zones, and smart furniture hookups.
     *
     * This method aggregates all spatial data into a single cohesive representation of the house.
     *
     * @return The complete house map with all its components
     *
     * @throws io.energyconsumptionoptimizer.mapservice.domain.errors.FlorPlanNotFoundException
     */
    suspend fun getHouseMap(): HouseMap
}

File: src/application/inbound/HouseMapService.ts
import type { HouseMap } from "@domain/entities/Map";

/**
 * @inbound
 * Service interface for retrieving the complete house map.
 */
export interface HouseMapService {
  /**
   * Retrieves the complete house map — floor plan, all zones, and all smart
   * furniture hookups — assembled into a single domain aggregate.
   *
   * @returns The HouseMap aggregate, or a FloorPlanNotFoundError if no floor
   *          plan has been uploaded yet.
   */
  getHouseMap(): Promise<HouseMap | Error>;
}
